### PR TITLE
More todo fixme cleanup

### PR DESCRIPTION
**Pull request by marcusnaslund**
Mostly changes to `open()` in `Raster*` classes. "Fixes" #561 
Removed commented-out code in `Synchronized` (we never use try-catch anywhere else, so...)
And fixed `TODO`s in `Vector`.

### CODE REVIEW

File: source/collections/Vector.ooc [hunk 1]
```diff
@@ -121,9 +120,7 @@ HeapVector: class <T> extends Vector<T> {
 
 	operator []= (index: Int, item: T) {
 		version (safe) {
-			// TODO: Should we also check that index < count?
-			// TODO: Should this affect count, if index > count?
-			if (index >= this capacity || index < 0)
+			if (index >= this capacity || index < 0 || index >= this capacity)
```
Comment (davidhesselbom) [inline source/collections/Vector.ooc]: Derp, you're checking `index >= this capacity` twice...

File: source/draw/RasterBgr.ooc [hunk 2]
```diff
@@ -105,14 +105,10 @@ RasterBgr: class extends RasterPacked {
 		}
 	}
 	open: static func (filename: String) -> This {
-		x, y, n: Int
+		x, y, totalComponents: Int
```
Comment (davidhesselbom) [inline source/draw/RasterBgr.ooc]: I think `imageComponents` would be a better name... don't you?

File: source/collections/Vector.ooc (hunk 1 above)
Comment (marcusnaslund) [inline source/collections/Vector.ooc]: If the computer miscomputes the first time, it's good to check again. :)

Comment (marcusnaslund) [inline source/collections/Vector.ooc]: Also: oops. The `TODO`s refer to `count` which doesn't even exist here. I didn't see the whole if-statement once I corrected for that.

Comment (davidhesselbom) [inline source/collections/Vector.ooc]: Right, `count` needs to be checked by `VectorList`.

File: source/draw/RasterBgr.ooc (hunk 2 above)
Comment (marcusnaslund) [inline source/draw/RasterBgr.ooc]: Hmm... yes.

**Comment by marcusnaslund** [comment]
Updated. 
`VectorList` now checks `count` (we don't need to check if it's `< 0` here).
Renamed `totalComponents` to `imageComponents`.

**Pull request merged**